### ISSUE TITLE
Check write permissions before attempting to download

### ISSUE THREAD

**Issue by smith** [habitat-sh/habitat#960]
When attempting to install as non-root to /hab:

```
nathansmith@opstop ~/habitat> hab pkg install core/python2
» Installing core/python2
→ Using core/acl/2.2.52/20160612075215
→ Using core/attr/2.4.47/20160612075207
→ Using core/bzip2/1.0.6/20160612075040
→ Using core/cacerts/2016.04.20/20160612081125
→ Using core/coreutils/8.24/20160612075329
→ Using core/gcc-libs/5.2.0/20160612075020
→ Using core/glibc/2.22/20160612063629
→ Using core/gmp/6.1.0/20160612064724
→ Using core/libcap/2.24/20160612075226
→ Using core/linux-headers/4.3/20160612063537
→ Using core/make/4.1/20160612080650
→ Using core/ncurses/6.0/20160612075116
→ Using core/openssl/1.0.2h/20160612081127
→ Using core/readline/6.3.8/20160612075601
→ Using core/zlib/1.2.8/20160612064520
↓ Downloading core/python2/2.7.11/20160620182552
    18.05 MB / 18.05 MB - [==================================================================================] 100.00 % 844.51 KB/s  ✗✗✗
✗✗✗ Can't create '/hab/pkgs/core/python2/2.7.11/20160620182552' (libarchive err_code=13)
✗✗✗
```

It would be nice if we checked our write permissions of this directory before downloading everything and failing early, instead of waiting until everything had downloaded.

**Comment by metadave** [comment]
Related: https://github.com/habitat-sh/habitat/issues/951

**Comment by metadave** [comment]
@Sgtpluck and I will be working on this issue on and off.

**Comment by arr-dev** [comment]
I'd like to take a stab at this if @metadave haven't worked on it?

**Comment by metadave** [comment]
it's ok with me, let's check with @Sgtpluck

**Comment by fnichol** [comment]
Oops, sorry all, I addressed this in #1325 indirectly. This feature prevents a non-root user from proceeding to run any commands that require root permissions, such as package installation in this case.

The new behavior for a non-root user would look like:

```
> hab install core/python2
∅ Installing a package requires root or administrator privileges. Please retry this command as a super user or use a privilege-granting facility such as sudo.

✗✗✗
✗✗✗ Root or administrator permissions required to complete operation
✗✗✗
```

---

How about this? I will try and find us another Hacktoberfest issue to replace this one.

Closed by #1325 

![gif-keyboard-8691805385755352059](https://cloud.githubusercontent.com/assets/261548/19122004/d7161860-8ae5-11e6-81ba-bef40c209e3b.gif)
